### ISSUE TITLE
More narrowly definition of what "function" is in context of 2.5.2 Pointer Cancellation

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#1542]
In reference to https://github.com/w3c/wcag/issues/1099 and https://github.com/w3c/wcag/issues/1529 I think there's some gray area when it comes to what a "function" is...

> For *functionality* that can be operated using a single pointer, at least one of the following is true

> Functionality: processes and outcomes achievable through user action

> Processes: series of user actions where each action is required in order to complete an activity

Now, it's fairly clear that if something substantial happens on a page (a new page is loaded, a modal is popped up, a form gets submitted, an action is initiated) that that counts as "functionality". However, what about milder cases like "the value on a widget changes (but nothing else is triggered)" (as is the case in #1529)? is that strictly a "functionality"? i'd argue that it's perhaps not - mainly because after an accidental triggering, it's trivially easy to undo the result here of the accidental triggering. if others agree, should this perhaps be explained a bit further in the understanding document/in the definition of "functionality"?

**Comment by alastc** [comment]
I would have thought that anything that changes a value would be considered functional, at least if it then contributes to a process. E.g. changing the number of items in a basket would.

Where the same control can "undo" or change the value back, that is excepted from the SC anyway.

**Comment by patrickhlauke** [comment]
> Where the same control can "undo" or change the value back, that is excepted from the SC anyway.

we've had this discussion before ... because of the normative wording, the "undo" is only allowed *IF* the completion is on the up event

> Abort or Undo: **Completion of the function is on the up-event**, *and* a mechanism is available to abort the function before completion or to undo the function after completion

**Comment by patrickhlauke** [comment]
> I would have thought that anything that changes a value would be considered functional

well this is what, I think, needs further thought and clarification. is changing a value (but the form itself isn't submitted or anything) "functionality"? is "functionality" distinct from triggering a particular action/function?

**Comment by steverep** [comment]
> well this is what, I think, needs further thought and clarification. is changing a value (but the form itself isn't submitted or anything) "functionality"? is "functionality" distinct from triggering a particular action/function?

I fully agree with @alastc here - certainly changing a value qualifies as functionality.  Every little independent thing a user can do with a page to produce an outcome is functionality.  Any vagueness there I believe serves WCAG well.  Narrowing that definition or trying to distinguish micro from macro functionality would just cause problems and loopholes in all the criteria that use the term, which is much more than 2.5.2.

For example, if changing a widget value is not functionality:
- Does it need to be keyboard operable?
- Does it still need to work if I resize text?
- Is it fine if I make it only work on mobile if I shake my device?
- Do I even need to worry about it being accessibility supported?

Again, as I mentioned in #1099, the mistake here is distinguishing any functionality based on how "trivial" or "mild" it may be.  What's trivial for one may not be for the next.

That said, I do believe there is some opportunity in Understanding for explaining functionality in the context of a criterion like 2.5.2 because it talks about "part of the function" and "completing the function" whereas other uses of the term do not.  The most clear example would be drag and drop.  The functionality is the outcome, i.e. the result of dropping (e.g. moving something, drawing something, etc.)  For 2.1.1, I just need to be able to produce that outcome with only a keyboard; I don't need to be able to simulate a drag and drop.  For 2.5.2, the single pointer method for achieving the outcome is to drag and drop, so the "Abort or Undo" clause must be met.

**Comment by patrickhlauke** [comment]
> For example, if changing a widget value is not functionality

that list now takes it to the extreme though. *everything* is functionality then. even scrolling is "functionality" then, and that will happen effectively on the down event for touch pointers. So again, this is more about being perhaps a bit more precise about what is/isn't in scope, as there's potential for the wording as currently written to have unintentionally strict applications (or just outright odd "yes, by default browsers do this on the down, but if YOU replicate that behaviour, you're failing" situations as mentioned in #1099 for when a `<select>` opens)
